vllm/model_executor/layers/fused_moe/layer.py

```diff
@@ -61,66 +55,45 @@ def create_weights(self, layer: torch.nn.Module, num_experts: int,
         layer.register_parameter("w2_weight", w2_weight)
         set_weight_attrs(w2_weight, extra_weight_attrs)
 
-    def apply(
-        self,
-        layer: torch.nn.Module,
-        x: torch.Tensor,
-        router_logits: torch.Tensor,
-        top_k: int,
-        renormalize: bool = True,
-        use_grouped_topk: bool = False,
-        num_expert_group: Optional[int] = None,
-        topk_group: Optional[int] = None,
-    ) -> torch.Tensor:
-        return self.forward(x, layer.w13_weight, layer.w2_weight,
-                            router_logits, top_k, renormalize,
-                            use_grouped_topk, num_expert_group, topk_group)
-
-    def forward_cuda(
-        self,
-        x: torch.Tensor,
-        w1: torch.Tensor,
-        w2: torch.Tensor,
-        router_logits: torch.Tensor,
-        top_k: int,
-        renormalize: bool,
-        use_grouped_topk: bool,
-        num_expert_group: Optional[int],
-        topk_group: Optional[int],
-    ) -> torch.Tensor:
-        from vllm.model_executor.layers.fused_moe.fused_moe import fused_moe
-        return fused_moe(x,
-                         w1,
-                         w2,
-                         router_logits,
-                         top_k,
-                         renormalize=renormalize,
-                         inplace=True,
-                         use_grouped_topk=use_grouped_topk,
-                         num_expert_group=num_expert_group,
-                         topk_group=topk_group)
+    def apply(self, layer: torch.nn.Module, x: torch.Tensor,
+              topk_weights: torch.Tensor, topk_ids: torch.Tensor,
+              **kwargs) -> torch.Tensor:
+
+        return self.forward(x=x,
+                            layer=layer,
+                            topk_weights=topk_weights,
+                            topk_ids=topk_ids,
+                            **kwargs)
+
+    def forward_cuda(self, layer: torch.nn.Module, x: torch.Tensor,
+                     topk_weights: torch.Tensor, topk_ids: torch.Tensor,
+                     **kwargs) -> torch.Tensor:
+        from vllm.model_executor.layers.fused_moe.fused_moe import (
+            fused_experts)
+        return fused_experts(hidden_states=x,
```

Comment:
It became a bit confusing that `FusedMoEMethodBase` actually doesn't do topk in `forward_cuda`, but does it in `forward_tpu`.

Reply:
We could potentially refactor fused_moe from moe_pallas in the same way we refactored fused_moe from fuse_moe.py but I think that's beyond the current scope of this PR

Reply:
I don't think that's beyond the scope of this PR, as now forward_cuda and other forwards are having different semantics which seem not a good practice we should have.

Reply:
Well either way the forward_tpu semantic will be different because of the way the topk weights are determined is different between the forward methods.

Reply:
We will move to have duplicate `select_expert` calls in all the `apply` quant methods so `FusedMoEMethodBase` always does topk, thanks for calling this out